docs/receipt-verifier.md

```diff
@@ -0,0 +1,98 @@
+
+## Install the receipt verifier service package
+
+You can get the [receipt verifier service package](https://github.com/coilhq/receipt-verifier) or run the verifier as a [Kubernetes service](https://github.com/coilhq/receipt-verifier/tree/main/config/base).
+
+**Example**
+
+We use Redis in this example. You can make a verifier that doesn't use Redis.
+
+```
+npm install
+npm run-script build
+sudo docker run -p 6379:6379 -d redis
+npm start
+```
+
+## Update your Web Monetization meta tag
+
+Typically, a Web Monetization meta tag looks something like this:
+
+`<meta name=“monetization" content="$wallet.example.com/alice">`
+
+Update your meta tag so that your URL-encoded payment pointer is in the path of the receipt verifier service’s URL.
+
+For example, if your payment pointer is `$wallet.example.com/alice` and your receipt verifier service’s URL is `https://receipt-verifier.com`, then you’ll set your meta tag’s `content` to either of the following:
+
+* `https://receipt-verifier.com/%24wallet.example.com%2Falice`
+* `$receipt-verifier.com/%24wallet.example.com%2Falice`
+
+With this change, any queries from the WM sender will be proxied by the receipt verifier service to `$wallet.example.com/alice`.
+
+## Web Monetization revshare generator
+
+If you used the [Web Monetization revshare generator](https://webmonetization.org/prob-revshare) to create a meta tag for [probabilistic revenue sharing](https://webmonetization.org/docs/probabilistic-rev-sharing/), follow the same instructions as above: URL-encode your revshare payment pointer and put it in the path of the receipt verifier service’s URL.
+
+For example, if your meta tag looks like this:
+
+`<meta name="monetization" content="$webmonetization.org/api/revshare/pay/W1siJHdhbGxldC5leGFtcGxlL2FsaWNlIiwxMCwiQWxpY2UiXSxbIiR3YWxsZXQuZXhhbXBsZS9ib2IiLDEwLCJCb2IiXV0">`
+
+And your receipt verifier service’s URL is `https://receipt-verifier.com`, then you’ll set your meta tag’s `content` to either of the following:
+
+* `https://receipt-verifier.com/%24webmonetization.org%2Fapi%2Frevshare%2Fpay%2FW1siJHdhbGxldC5leGFtcGxlL2FsaWNlIiwxMCwiQWxpY2UiXSxbIiR3YWxsZXQuZXhhbXBsZS9ib2IiLDEwLCJCb2IiXV0`
+* `$receipt-verifier.com/ %24webmonetization.org%2Fapi%2Frevshare%2Fpay%2FW1siJHdhbGxldC5leGFtcGxlL2FsaWNlIiwxMCwiQWxpY2UiXSxbIiR3YWxsZXQuZXhhbXBsZS9ib2IiLDEwLCJCb2IiXV0`
+
+## Set up a monetizationprogress event listener
+
+When a WM sender receives a receipt, the sender notifies the user agent of the successful payment by submitting a `monetizationprogress` event containing a receipt.
+
+Add the following client-side code to your website to listen for a `monetizationprogress` event containing a `receipt`. When a receipt exists, your website will send it to the receipt verifier in the event listener to confirm the payment.
```

Comment:
we should rethink this example; the security isn't actually improved by speaking directly to the receipt verifier from the frontend, because anyone could change what code is on the frontend or intercept the request to make it look like the request succeeded. The proper way to use the receipt verifier would be to submit a receipt to your app's backend and then have your app's backend submit the receipt to the verifier in order to ensure it's validity. Then the app backend can securely take action like returning exclusive content or giving some perk to the user's account.